grafana-plugin/src/containers/ServiceNowConfigDrawer/ServiceNowTokenSection.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, HorizontalGroup, LoadingPlaceholder, VerticalGroup, useStyles2 } from '@grafana/ui';
+import { observer } from 'mobx-react';
+
+import { IntegrationInputField } from 'components/IntegrationInputField/IntegrationInputField';
+import { RenderConditionally } from 'components/RenderConditionally/RenderConditionally';
+import { Text } from 'components/Text/Text';
+import { ActionKey } from 'models/loader/action-keys';
+import { useCurrentIntegration } from 'pages/integration/OutgoingTab/OutgoingTab.hooks';
+import { useStore } from 'state/useStore';
+import { useIsLoading } from 'utils/hooks';
+
+import { getCommonServiceNowConfigStyles } from './ServiceNow.styles';
+
+
+export const ServiceNowTokenSection: React.FC = observer(() => {
+  const styles = useStyles2(getStyles);
+  const { id } = useCurrentIntegration();
+  const { alertReceiveChannelStore } = useStore();
+  const [isExistingToken, setIsExistingToken] = useState(undefined);
+  const [currentToken, setCurrentToken] = useState(undefined);
+  const isLoading = useIsLoading(ActionKey.UPDATE_SERVICENOW_TOKEN);
+
+  useEffect(() => {
+    (async function () {
+      const hasToken = await alertReceiveChannelStore.hasServiceNowToken({ id });
```

Comment:
I suggest that we:
1. Rename `hasServiceNowToken` into `checkIfServiceNowHasToken`
2. Create small boolean observable called `serviceNowHasToken` in `alertReceiveChannelStore` that `checkIfServiceNowHasToken` will update upon receiving response from the backend
3. Remove `isExistingToken` local state and use new `serviceNowHasToken` from the store directly

I already have another place that can reuse it, we have a duplication now

Reply:
Done, moved to helpers & renamed it. I'm a bit hesitant into adding more data to the alert receive channel store because it's not the kind of data that is largely used by other `alert_receive_channel` consumers... my preference would be to have it in the component because store-wise and being consumed by just 1 component seems kinda meh (maybe there will be 1 or 2 more but the next point still stands). Add to that the store shouldn't know about ServiceNow stuff... Imagine we will add another integrations, it would be nightmare to have so maintain such small data for each integration in the store

Reply:
Got it, helper is ok. As mentioned in previous comment, I'm just trying to point our attention to a bigger picture which is - it would be awesome if we try to make our view layer - components - simpler, flatter etc. There are different ways to achieve that - custom hooks, helpers, better components composition etc

I can already see in your PRs that you pay attention to it and thank you for that! Just giving a bit more suggestions, I hope that's ok